### PR TITLE
add stackstorm-rbac volume to st2api and st2client

### LINKED ISSUE #219: 3.4 rbac drive addition
**Is your feature request related to a problem? Please describe.**
With 3.4 arriving I've been testing the dev version in docker. RBAC is included in this version but requires files (roles, users, etc) in order for this function. Right now there is no place to host and manage these files for the containers that need them.

**Describe the solution you'd like**
I was thinking maybe it would be a good idea to add a static rbac volume to host all these files and shared between the relevant containers. I just dont know what containers needs to access these files outside of the auth container.

If these containers are identified I'm happy to push a 3.4 branch with these changes.

**Describe alternatives you've considered**
The other alternative is outside the scope of this repo which is to add a proper interface component to manage rbac features.

### PR DESCRIPTION
**Pull request by minsis**
closes #219

### CODE REVIEW

**Review by armab — CHANGES_REQUESTED** [review]
Thanks for the PR!
Adding RBAC volume is a good start for enabling previously enterprise functionality.

What's the user's flow to configure the RBAC/LDAP in full from the st2-docker environment? Do we have documentation/automation/scripts for that? We want the full feature here.

**Comment by minsis** [comment]
Not sure I fully understand the ask here. All documentation on how to configure RBAC is in the StackStorm official documentation and doesn't differ from the docker version of this setup. I did add to the gotchas that since there is a st2.docker.conf that you need to specify that as the config file when trying to run the RBAC update script. Thats really the only difference I can think of unless I'm missing something.

**Comment by armab** [comment]
The main point is that just enabling RBAC volume is not sufficient to call this functionality usable and clear from the user's perspective.

For instance, in K8s environment, the RBAC feature is enabled via a flag and we ship the default RBAC roles and assignments https://github.com/StackStorm/stackstorm-ha/blob/master/values.yaml#L111-L145.

Additionally, similar to automatic st2 pack load for docker deployment here, do you think of any helpers around `st2-apply-rbac-definitions` that might simplify user experience? What do you think would be more helpful comparing to what we have now?

**Comment by minsis** [comment]
Since this setup doesn't exactly have something like helm to do the build we're relying solely on docker-compose and afaik there isn't anything like what helm can do with values.yaml... maybe?

Some thoughts off the top of my head:
* Create a script that would determine if RBAC is enabled from the config and then process from there.
  * this might be the simplest solution, but might not be what we want for any future expansion or plans.
* Specify an environment variable in something like values.env and said script would process from there. This might be a building block for something like the k8s setup.
  * This is doable but sort of feels messy with adding a bunch of flags and/or configs as environment variables inside containers. Especially if you need to provide sensitive information.
* Lastly, create an actual values.yaml file and build a customized script that processes it.
  * This would be most beneficial for any future plans
  * Downside is we might now be asking the user to manage two config files (st2.user.conf and values.yaml), unless we now replace st2.user.conf with pre-defined options from values.yaml

Since `st2-apply-rbac-definitions` is just a wrapper around some python modules it would fairly simple to come up with some type of processing script on startup. I think maybe the only question on any of this is do we process everything at start up or at 'build' time.

Thoughts or ideas?

**Comment by armab** [comment]
At a minimum, I think it would be nice to include the [default roles and assignments](https://github.com/StackStorm/stackstorm-ha/blob/master/values.yaml#L111-L145) in the `stackstorm-rbac` directory in this repo.
With that, at least users will have to start with the known good configuration and we may provide further recommendations in the README.

At maximum, running `st2-apply-rbac-definitions` as part of a startup inside the st2client, similar to what happens with the register content.

File: files/rbac/mappings/stormers.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# StackStorm RBAC LDAP groups-to-roles mapping rules, shipped in '/opt/stackstorm/rbac/mappings/'
+# See RBAC Roles Based on LDAP Groups: https://docs.stackstorm.com/rbac.html#automatically-granting-roles-based-on-ldap-group-membership
+---
+group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
+description: "Automatically grant admin role to all stormers group members."
+roles:
+  - "admin"
```
Comment (armab) [inline files/rbac/mappings/stormers.yaml:7]:
```suggestion
# ---
# group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
# description: "Automatically grant admin role to all stormers group members."
# roles:
#   - "admin"
```

Just one note to comment out the LDAP < > RBAC mappings as that's not the default configuration and will blow up without the LDAP auth settings in st2.conf.

Besides that, looks good. Great work!

Comment (minsis) [reply files/rbac/mappings/stormers.yaml:7]: Yes I already tried with them all commented out, however `st2-apply-rbac-definitions` fails to execute the import when there's a file with nothing in it. I ran the sync with that file uncommented and it runs fine.

```shell
root@49f0bd852fa5:/opt/stackstorm# cat rbac/mappings/stormers.yaml 
# StackStorm RBAC LDAP groups-to-roles mapping rules, shipped in '/opt/stackstorm/rbac/mappings/'
# See RBAC Roles Based on LDAP Groups: https://docs.stackstorm.com/rbac.html#automatically-granting-roles-based-on-ldap-group-membership
#---
#group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
#description: "Automatically grant admin role to all stormers group members."
#roles:
#  - "admin"
root@49f0bd852fa5:/opt/stackstorm# st2-apply-rbac-definitions --config-file /etc/st2/st2.docker.conf
2021-04-09 22:59:41,484 INFO [-] Connecting to database "st2" @ "mongo:27017" as user "None".
2021-04-09 22:59:41,495 INFO [-] Successfully connected to database "st2" @ "mongo:27017" as user "None".
2021-04-09 22:59:41,943 INFO [-] Loading role definitions from "/opt/stackstorm/rbac/roles/"
2021-04-09 22:59:42,086 INFO [-] Loading user role assignments from "/opt/stackstorm/rbac/assignments/"
2021-04-09 22:59:42,096 INFO [-] Loading group to role map definitions from "/opt/stackstorm/rbac/mappings/"
Traceback (most recent call last):
  File "/usr/bin/st2-apply-rbac-definitions", line 15, in <module>
    sys.exit(apply_rbac_definitions.main(sys.argv[1:]))
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/cmd/apply_rbac_definitions.py", line 60, in main
    apply_definitions()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/cmd/apply_rbac_definitions.py", line 44, in apply_definitions
    result = loader.load()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 65, in load
    result['group_to_role_maps'] = self.load_group_to_role_maps()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 136, in load_group_to_role_maps
    file_path=file_path)
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 191, in load_group_to_role_map_assignment_from_file
    raise ValueError(msg)
ValueError: Group to role map assignment file "/opt/stackstorm/rbac/mappings/stormers.yaml" is empty and invalid
```

How do you want to handle this?

Comment (armab) [reply files/rbac/mappings/stormers.yaml:7]: Thanks for trying that!
I'm fine with removing the example mappings file then.

**Review by armab — APPROVED** [review]
Thanks for the changes integrating this feature :+1:

**Pull request merged**